docs/contributing.md

```diff
@@ -0,0 +1,347 @@
+
+With the exception of code blocks, limit line length to 79 characters. You may
+want to use your editor's tools to automatically hard wrap lines of text.
+
+Don't use abbreviations such as 'e.g.' but instead use the long form, such as
+'For example'.
+
+The documentation is built from the [Markdown] source files in the [docs
+directory] by the [MkDocs] static site generator. In addition to the basic
+Markdown syntax, the following extensions are supported: [extra], [admonition],
+[smarty], [codehilite], and [toc].
+
+There are a few conventions you should follow when working on the
+documentation.
+
+#### Headers
+
+Headers should use the hash style. For example:
+
+```md
+## Some important topic
+```
+
+The underline style should not be used. Don't do this:
+
+```md
+Some important topic
+====================
+```
+
+#### Links
+
+Links should always use the reference style, with the referenced hyperlinks kept
+at the end of the document.
+
+```md
+Here is a link to [some other thing][other-thing].
+
+More text...
+
+[other-thing]: http://example.com/other/thing
+```
+
+This style helps keep the documentation source consistent and readable.
+
+If you are linking to another document within Python-Markdown's documentation,
+you should use a relative link, and link to the `.md` suffix. If applicable, it
+is preferred that the link includes a hash fragment pointing to the specific
+section of the page. For example:
+
+```md
+[authentication]: reference.md#Markdown
+```
+
+Linking in this style ensures that the links work when browsing the
+documentation on GitHub. If your Markdown editor makes links clickable, they
+will work there as well. When the documentation is built, these links will be
+converted into regular links which point to the built HTML pages.
+
+#### Notes and Warnings
+
+If you want to draw attention to a note or warning, use the syntax defined in
+Python-Markdown's [Admonition Extension]:
+
+```md
+!!! note
+
+    This is the content of the note.
+```
+
+### Commit Message Style Guide
+
+Use the present tense ("Add feature" not "Added feature").
+
+Use the imperative mood ("Move item to..." not "Moves item to...").
+
+Limit the first line to 72 characters or less.
+
+Reference issues and pull requests liberally after the first line. Include a
+summary of the changes/additions made without replicating the content of the
+documentation or release notes. This is where an explanation of the choices made
+should be found. References to issues and pull requests should only provide the
+context in which a choice was made. However, the commit should be able to stand
+on its own.
+
+## Versions
+
+Python-Markdown follows [Semantic Versioning] and uses the
+`MAJOR.MINOR.POINT(PRERELEASE/DEV)` format for specifying releases. The status
+of the `master` branch should always be identified in the `__version_info__`
+variable defined in [`markdown/__init__.py`][markdown/__init__.py] and should
+conform to [PEP 440].
```

Comment:
Just a note that [pypa/packaging] officially implements PEP 404. And as of pypa/setuptools@9382fa0c, (version 8.0) `setuptools` and `pkg_resources` vendor and use `pypa/packaging` to do all version parsing. Finally, since 52b8da8e (version 3.0) we are using `setuptools` for installation.

Therefore, the differences between `pkg_resources` and PEP 440 listed in the PEP are presumably outdated and we should be able to use `pkg_resources` for all version handling without any added dependencies. In other words, we can remove our own version parsing code when conforming to PEP 440.

[pypa/packaging]: https://github.com/pypa/packaging/